src/Elliptic/Systems/Elasticity/BoundaryConditions/LaserBeam.cpp

```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/Elasticity/BoundaryConditions/LaserBeam.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryCondition.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Elasticity::BoundaryConditions::detail {
+
+void LaserBeamImpl::apply(
+    const gsl::not_null<tnsr::I<DataVector, 3>*> /*displacement*/,
+    const gsl::not_null<tnsr::I<DataVector, 3>*> n_dot_minus_stress,
+    const tnsr::I<DataVector, 3>& x,
+    const tnsr::i<DataVector, 3>& face_normal) const noexcept {
+  const auto n_dot_x = dot_product(face_normal, x);
+  const auto r_sq = square(get<0>(x)) + square(get<1>(x)) + square(get<2>(x)) -
+                    square(get(n_dot_x));
+  const DataVector beam_profile =
+      exp(-r_sq / square(beam_width_)) / (M_PI * square(beam_width_));
+  get<0>(*n_dot_minus_stress) = -beam_profile * get<0>(face_normal);
```

Comment:
If this function is run often enough to want to avoid allocations altogether, you can do so without much pain by writing the to the last element of the array instead.

Reply:
Inhomogeneous boundary conditions are actually only evaluated _once_ in the elliptic solve, so I think I'll leave this simpler code :)